crates/ignore/src/default_types.rs

```diff
@@ -290,3 +290,22 @@ pub const DEFAULT_TYPES: &[(&str, &[&str])] = &[
     ]),
     ("zstd", &["*.zst", "*.zstd"]),
 ];
+
+#[test]
+fn test_default_types_are_sorted() {
+    let mut names = DEFAULT_TYPES.iter().map(|(name, _exts)| name);
+
+    if let Some(first) = names.next() {
```

Comment:
I would write this as:

```rust
let first = match names.next() {
    None => return,
    Some(first) => first,
};
```

Or perhaps even better, using the [newly stabilized `let else`](https://github.com/rust-lang/rust/pull/93628/):

```rust
let Some(first) = names.next() else { return };
```

Although in order to use `let else` we need to bump MSRV to Rust 1.65. Since ripgrep tracks the latest stable release of Rust, I have no problem with that. Indeed, [I just pushed a commit to do so](https://github.com/BurntSushi/ripgrep/commit/8905d54a9f25f4c1e4e3ca8331f517473e174d87), so all you need to do to use `let else` is to rebase on master.

Reply:
The higher level point here is to keep code linear via early returns. That way, you avoid nesting the main body here.

Reply:
That's much neater. Of course the best solution would be rust-lang/rust#53485, but that seems to still be quite a while away.

If ripgrep uses latest stable anyway, what's the opinion on `edition = "2021"`? It would allow for things like implicit arguments in panic/assert format strings: https://doc.rust-lang.org/edition-guide/rust-2021/panic-macro-consistency.html.

Reply:
I'll very likely do the edition migration myself before the next release, whenever that is. I'm in no rush to be honest. I'll likely only do things that `cargo fix` does automatically. Converting explicit to implicit arguments by hand is a fair bit of work.

With respect to `is_sorted`, I actually think your approach here is better. It gives a better failure mode by providing a witness. `is_sorted` wouldn't do that. :-)